### PR TITLE
Support parsing into vector in json

### PR DESCRIPTION
**Pull request by data-man**

### CODE REVIEW

File: lib/std/json.zig [hunk 1]
```diff
@@ -1531,6 +1531,40 @@ fn parseInternal(comptime T: type, token: Token, tokens: *TokenStream, options:
...
+                        r[i] = try parse(vectorInfo.child, tokens, options);
+                    }
+                    const tok = (try tokens.next()) orelse return error.UnexpectedEndOfJson;
+                    switch (tok) {
+                        .ArrayEnd => {},
+                        else => return error.UnexpectedToken,
+                    }
+                    return r;
+                },
+                .String => |stringToken| {
```
Comment (daurnimator) [inline lib/std/json.zig:1555]: Do we really want to parse a string into a vector?

File: lib/std/json.zig [hunk 2]
```diff
@@ -1624,6 +1658,11 @@ pub fn parseFree(comptime T: type, value: T, options: ParseOptions) void {
                 parseFree(arrayInfo.child, v, options);
             }
         },
+        .Vector => |vectorInfo| {
```
Comment (daurnimator) [inline lib/std/json.zig:1661]: could combine this with the `Array` case above.

Comment (data-man) [reply lib/std/json.zig:1661]: I tried `.Array, .Vector => |info| {`).
and got an `error: capture group with incompatible types'.

File: lib/std/json.zig (hunk 1 above)
Comment (data-man) [reply lib/std/json.zig:1555]: Why not?
It will allow fast string processing in future. :)

**Comment by andrewrk** [comment]
> It may be better to leave this unimplemented to provide such guidance than to make it work, even though it is technically possible.

Closing because of this. However please re-open if you would like to continue to make the case to include this change.

**Pull request closed without merging**